api/src/main/java/jakarta/data/metamodel/Expression.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.metamodel.constraint.Constraint;
+import jakarta.data.metamodel.restrict.Restriction;
+
+import java.util.Set;
+
+public interface Expression<T,V> {
+
+    Restriction<T> equalTo(V value);
+    Restriction<T> notEqualTo(V value);
```

Comment:
```suggestion
    Restriction<T> equalTo(Expression<T, V> expression);
    Restriction<T> equalTo(V value);
    Restriction<T> notEqualTo(Expression<T, V> expression);
    Restriction<T> notEqualTo(V value);
```

so that you can compare two attributes to each other:

```
        Restriction<Book> autobiography = _Book.title.equalTo(_Book.author);
```

Reply:
Was just thinking/wondering if it's more complicated than that, but maybe not, perhaps that's all that's needed.

Reply:
> thinking/wondering if it's more complicated than that

So the complications arise when you ask if these sort of cross-field restrictions are made from `Constraint`s. After some reflection, I don't think there are. They would be a new sort of `Restriction`. That's actually fine, but it's probably not something we want to bite off at first (it can easily be added).